homeassistant/components/islamic_prayer_times/sensor.py

```diff
@@ -16,26 +16,28 @@ async def async_setup_entry(
 ) -> None:
     """Set up the Islamic prayer times sensor platform."""
 
-    client = hass.data[DOMAIN]
+    coordinator = hass.data[DOMAIN]
 
     entities = []
     for sensor_type in SENSOR_TYPES:
-        entities.append(IslamicPrayerTimeSensor(sensor_type, client))
+        entities.append(IslamicPrayerTimeSensor(sensor_type, coordinator))
 
-    async_add_entities(entities, True)
+    async_add_entities(entities)
 
 
-class IslamicPrayerTimeSensor(SensorEntity):
+class IslamicPrayerTimeSensor(
+    CoordinatorEntity[IslamicPrayerDataUpdateCoordinator], SensorEntity
+):
     """Representation of an Islamic prayer time sensor."""
 
     _attr_device_class = SensorDeviceClass.TIMESTAMP
     _attr_icon = PRAYER_TIMES_ICON
     _attr_should_poll = False
 
-    def __init__(self, sensor_type, client):
+    def __init__(self, sensor_type, coordinator):
```

Comment:
Is it possible to add type hints?
```suggestion
    def __init__(self, sensor_type: str, coordinator: IslamicPrayerDataUpdateCoordinator) -> None:
```

Reply:
Done